.travis.yml

```diff
@@ -21,13 +21,14 @@ before_install:
   - if [[ $TRAVIS_PYTHON_VERSION < '3.0' ]]; then pip install mock; fi
   - pip install codecov
   - pip install freezegun
+  - pip install pytest
 
 install:
   - pip install six
   - ./ci_tools/retry.sh python updatezinfo.py
 
 script:
-  - coverage run --omit=setup.py,dateutil/test/* setup.py test
+  - coverage run --omit=setup.py,dateutil/test/* -m pytest
```

Comment:
Let's separate this out, as it has caused heisenbugs in the past.

Reply:
Separate it out how?

Reply:
Also, if by "heisenbugs" you mean that issue with your other PR, I think that *may* have been to do with the `assert` method of creating tests.

Reply:
> Separate it out how?

I'm suggesting the move to pytest should be done in an independent PR (I think you may have already updated this below)  

> may have been to do with the yield method of creating tests

Good to know, thanks.

Reply:
OK, yeah, maybe I'll cherry-pick the pytest commit and rebase this on master after we do that.

Reply:
If you're doing cherry-picking for small fairly-immediate changes, making `parser` a directory is worthwhile.  It'll make for much smaller diffs in other PRs.

Reply:
Hm.. Probably a good idea, but I think I'll just wrap up the feature as is and do one big rebase at the end.